### ISSUE TITLE
Bug using preserve color option

### ISSUE THREAD

**Issue by borderlineinteractive** [titu1994/Neural-Style-Transfer#70]
Hi,

Thank you very much for this nice implementation.

I am using your code successfully on Windows 10 on an iMac and AMD GPU via Plaidml. However, using the preserve color option I get a noise-like Image output with a diagonal stripe pattern. This also happens with the example images that you provide and show on your GitHub page (the Buddhist temple with water lillies style. Without the preserve color option everything works fine.

Any suggestion what might be going wrong?

Many thanks in advance!

**Comment by titu1994** [comment]
There's been some issue to get color preservation working once scipy stopped supporting imread and imsave. I'd suggest going down to a version which uses older scipy and perform colorization post training

**Comment by sipie800** [comment]
I encounter this issue too using Tensorlow 2.3 Keras 2.4.3 and lastest other requirement modules up to posting date.
And I finally figured out that this is due to some implementation using  PIL and skimage.Generally these two libraries is not-so-good if concerning rigorous projec implementing.
The color preservation use colorspace convertion back and forth between RGB and YCbCr.They use skimage.color.rgb2ycbcr and .ycbcr2rgb.But the 2 function are badly documented that they seems to misunderstand how to use it.For example when using .ycbcr2rgb the result is [0.0,1.0] not [0-255].Old version scipy.misc may handle this because they are coupled libraies.However newer imageio or PIL can't.So we get noisy image by using newer imageio or PIL.And PIL itself has some issues too.
Using cv2 get it work:
Network.py
def original_color_transform(content, generated, mask=None):
...
#RGB->BGR->YCbCr
cv2.cvtColor(generated[:, :, ::-1], cv2.COLOR_BGR2YCR_CB)
...
#RGB<-BGR<-YCbCr
generated=cv2.cvtColor(generated,cv2.COLOR_YCR_CB2BGR)[:, :, ::-1]
...
And don't need utils.fromimage() and .toimage() anymore.

**Comment by sipie800** [comment]
I encounter this too.Win10 tensorflow 2.3 and Keras 2.4.3 and all latest version required libraries.
Finally figured out that this is due to some glitch on conversion between rgb and ycbcr and also between uint8 and float64,
and al the same for working around among PIL,skimage,imageio.
It's not a wise choose to fix all the glitches one by one because those libraries are not designed to work together in  a decent way.
Get it solved by removing all implementation using PIL,imageio and skimage.All of these will be done by cv2 only.
And also conversion between RGB and BGR is done by cv2 rather than x=x[:,:,::-1], which just in some case, can cause unexpected result where you never imagine to be.

**Comment by HitLuca** [comment]
I know I'm late to the party, but to fix it you can update the `fromimage` function in `utils.py` from 

```python
def fromimage(img, mode="RGB"):
    if mode == "RGB":
        img = color.lab2rgb(img)
    else:
        img = color.rgb2lab(img)
    return img
```

to

```python
def fromimage(img, mode="RGB"):
    return np.array(img.convert(mode))
```

as a bonus you don't need the `scikit-image` library anymore since it's used only for that from what I could see
